migrator/migrations/m_169_to_m_170_move_scope_to_collection_in_report_configurations/migration.go

```diff
@@ -0,0 +1,229 @@
+				Value:     cluster,
+				MatchType: storage.MatchType_EXACT,
+			})
+		}
+		col := buildEmbeddedCollection(scope.GetName(), index, []*storage.SelectorRule{
+			{
+				FieldName: search.Cluster.String(),
+				Operator:  storage.BooleanOperator_OR,
+				Values:    ruleVals,
+			},
+		})
+		collectionsToEmbed = append(collectionsToEmbed, col)
+		index = index + 1
+	}
+
+	if includedNamespaces := scope.GetRules().GetIncludedNamespaces(); len(includedNamespaces) > 0 {
+		for _, namespace := range includedNamespaces {
+			col := buildEmbeddedCollection(scope.GetName(), index, []*storage.SelectorRule{
+				{
+					FieldName: search.Cluster.String(),
+					Operator:  storage.BooleanOperator_OR,
+					Values: []*storage.RuleValue{
+						{
+							Value:     namespace.GetClusterName(),
+							MatchType: storage.MatchType_EXACT,
+						},
+					},
+				},
+				{
+					FieldName: search.Namespace.String(),
+					Operator:  storage.BooleanOperator_OR,
+					Values: []*storage.RuleValue{
+						{
+							Value:     namespace.GetNamespaceName(),
+							MatchType: storage.MatchType_EXACT,
+						},
+					},
+				},
+			})
+			collectionsToEmbed = append(collectionsToEmbed, col)
+			index = index + 1
+		}
+	}
+
+	if clusterLabelSelectors := scope.GetRules().GetClusterLabelSelectors(); len(clusterLabelSelectors) > 0 {
+		labelCollections, success := labelSelectorsToCollections(scope.GetName(), index, clusterLabelSelectors, search.ClusterLabel.String())
+		if !success {
+			return nil, false
+		}
+		index = index + len(labelCollections)
+		collectionsToEmbed = append(collectionsToEmbed, labelCollections...)
+	}
+
+	if namespaceLabelSelectors := scope.GetRules().GetNamespaceLabelSelectors(); len(namespaceLabelSelectors) > 0 {
+		labelCollections, success := labelSelectorsToCollections(scope.GetName(), index, namespaceLabelSelectors, search.NamespaceLabel.String())
+		if !success {
+			return nil, false
+		}
+		collectionsToEmbed = append(collectionsToEmbed, labelCollections...)
+	}
+	return collectionsToEmbed, true
+}
+
+func moveScopeIDToCollectionIDInReports(db *pgxpool.Pool) error {
+	if !features.ObjectCollections.Enabled() {
+		return nil
+	}
+	ctx := context.Background()
+	reportConfigStore := reportConfigurationPostgres.New(db)
+	accessScopeStore := accessScopePostgres.New(db)
+	collectionStore := collectionPostgres.New(db)
+
+	reportConfigsToUpsert := make([]*storage.ReportConfiguration, 0, batchSize)
+	err := reportConfigStore.Walk(ctx, func(reportConfig *storage.ReportConfiguration) error {
+		scopeID := reportConfig.GetScopeId()
+		scope, found, err := accessScopeStore.Get(ctx, scopeID)
+		if err != nil {
+			return errors.Wrapf(err, "error migrating scope used in report configuration %s: failed to fetch scope id %s", scopeID, reportConfig.GetName())
+		}
+		if !found {
+			return errors.Errorf("error migrating scope used in report configuration %s: scope id %s not found", reportConfig.GetName(), scopeID)
```

Comment:
Do not error out. We still should process other configs. Log error and continue.